assets/app/actions/siteActions.js

```diff
@@ -65,7 +66,15 @@ export default {
 
   fetchSiteConfigsAndAssets(site) {
     return this.fetchSiteConfigs(site).then((site) => {
-      return this.fetchSiteAssets(site);
+      return this.fetchSiteAssets(site).then((site) => {
```

Comment:
> but my feeling is that its necessary when they enter the editor or click the 'Media' link

That is accurate.

> One request should be able to get the _config.yml, _navigation.json, and general content of the site correct?

I don't think so. From what I can tell ([Github API docs](https://developer.github.com/v3/repos/contents/#response-if-content-is-a-directory)), the response from an endpoint like `https://api.github.com/repos/:user/:repo/contents` returns an array of the root contents, but a request is still needed to get the actual content of that file.